ingest/scripts/override_group_segments.py

```diff
@@ -185,12 +185,15 @@ def write_grouped_metadata(
     different_values_log = {}
     count_total = 0
     count_ungrouped = 0
-    for record in orjsonl.stream(input_metadata_path):
+    for record in orjsonl.stream(input_metadata_path):  # type: ignore
+        if not isinstance(record, dict):
+            error = f"Expected a dict, got {type(record)} in {input_metadata_path}"
+            raise TypeError(error)
         count_total += 1
         metadata = record["metadata"]
         if metadata["insdcAccessionFull"] not in groups.accession_to_group:
             count_ungrouped += 1
-            orjsonl.append(
+            orjsonl.append(  # type: ignore
```

Comment:
I would think there's a better way to ignore mypy errors than to add `# type: ignore` to every single use of it - that's a bit annoying to read and maintain